hack/generator/pkg/codegen/code_generator.go

```diff
@@ -89,17 +95,21 @@ func createAllPipelineStages(idFactory astmodel.IdentifierFactory, configuration
 		nameTypesForCRD(idFactory),
 
 		// Apply property type rewrites from the config file
-		// must come after nameTypesForCRD and convertAllOfAndOneOf so that objects are all expanded
-		applyPropertyRewrites(configuration),
+		// Must come after nameTypesForCRD ('nameTypes)' and convertAllOfAndOneOfToObjects ('allof-anyof-objects') so
+		// that objects are all expanded
+		applyPropertyRewrites(configuration).
+			RequiresPrerequisiteStages("nameTypes", "allof-anyof-objects"),
 
 		// Figure out ARM resource owners:
 		determineResourceOwnership().UsedFor(ArmTarget),
 
 		// Strip out redundant type aliases:
 		removeTypeAliases(),
 
-		// De-pluralize resource types:
-		improveResourcePluralization(),
+		// De-pluralize resource types
+		// (Must come after type aliases are resolved)
+		improveResourcePluralization().
+			RequiresPrerequisiteStages("removeAliases"),
```

Comment:
rather than coding in the names here it would probably be better to do e.g. 

```golang
removeAliases := removeTypeAliases()
// ... 
improveResourcePluralization().RequiresPrerequisiteStages(removeAliases.Name())
```